storage/database/shardedPersister.go

```diff
@@ -0,0 +1,154 @@
+package database
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/sharding"
+	"github.com/multiversx/mx-chain-go/storage"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-chain-storage-go/leveldb"
+)
+
+var log = logger.GetOrCreate("storage/database")
+
+type persisterIDProvider interface {
+	ComputeIdFromBytes(address []byte) uint32
+}
+
+type shardedPersister struct {
+	persistersMut    sync.RWMutex
+	persisters       map[uint32]storage.Persister
+	shardCoordinator sharding.Coordinator
+}
+
+func NewShardedPersister(path string, batchDelaySeconds int, maxBatchSize int, maxOpenFiles int, shardCoordinator sharding.Coordinator) (*shardedPersister, error) {
+	persisters := make(map[uint32]storage.Persister)
+	for _, shardID := range getShardIDs(shardCoordinator) {
+		newPath := updatePathWithShardID(path, shardID)
+		db, err := leveldb.NewDB(newPath, batchDelaySeconds, maxBatchSize, maxOpenFiles)
+		if err != nil {
+			return nil, err
+		}
+		persisters[shardID] = db
+	}
+
+	return &shardedPersister{
+		persisters:       persisters,
+		shardCoordinator: shardCoordinator,
+	}, nil
+}
+
+func updatePathWithShardID(path string, shardID uint32) string {
+	if shardID == core.MetachainShardId {
+		return fmt.Sprintf("%s_%s", path, "Meta")
+	}
+
+	return fmt.Sprintf("%s_%d", path, shardID)
+}
+
+func getShardIDs(shardCoordinator sharding.Coordinator) []uint32 {
+	shardIDs := make([]uint32, shardCoordinator.NumberOfShards()+1)
+	for i := uint32(0); i < shardCoordinator.NumberOfShards(); i++ {
+		shardIDs[i] = i
+	}
+	shardIDs[shardCoordinator.NumberOfShards()] = core.MetachainShardId
```

Comment:
no need to add metachain to the DB shards

Reply:
done